### ISSUE TITLE
Attachment lost on re-open of a log or game file

### ISSUE THREAD

**Issue by riverwanderer** [vassalengine/vassal#13315]
During development of the C&C Napoleonics v5 module, I noticed that a couple of Attachments are lost when a log file is opened.

The Attachments are formed when the game starts up, with a trait of the form:
<img width="948" alt="image" src="https://github.com/vassalengine/vassal/assets/58135975/41fb4cef-5a08-442c-810c-839b7eb4c034">

The Sidebar uses this Attachment to determine which national flag to display for each side during Combat Assist operations. 

The Attachment traits are found in the prototype `Combat Assist Side Bar`. The last release using this mechanism is beta8, available [here](https://www.dropbox.com/scl/fi/3e8wiy5muzfprufo1othv/CCN5beta8demo.vmod?rlkey=ruw2uqpolwtjyvo8s1hpuc2vd&dl=0) (Dropbox link). The issue is demonstrated by these steps:
1. Open a non-British scenario, such (any from Russian, Austrian or Prussian expansions). 
2. Start a game with a log file, play through until a combat has occurred during which a flag is displayed on the Combat Assist sidebar for the upper side. Close the log file.
3. Now, re-open the log file and play through to the same point. Notice that a British flag is now displayed.
4. On checking the `CA Side Bar` piece property `upper_AttachCount,` it is zero from the start of the re-opened log file. Same for `lower_AttachCount`, but here the default is French flag, so that does not demo the issue.
5. If the Sidebar is re-initialised (e.g. by a test GKC sending the `initMarker` Key Command), then the flag will display correctly again.

Workaround that I am adopting for subsequent versions is to do a one-time setup, forming the attachments and saving required data in Dynamic Properties.

I am not sure that this is a general issue for all Attachments. The Sidebar also attaches to the CA Controller piece and other pieces, but these Attachment seem to be preserved, all but one are Auto-Attachments. The other is less used and possibly needs further testing.

**Comment by riverwanderer** [comment]
Example from the Iron will manifestation described above... the attached game file will open with this piece (context menu shown):
<img width="209" alt="image" src="https://github.com/vassalengine/vassal/assets/58135975/586589d9-87ef-4f2b-ade9-af9693e4845a">

In fact the menu should also shown a Ctrl+B command but it is suppressed because the attachment `ca` to the Retreat widget (flag), formed before the game was saved, is no longer connected. The debugger reveals this:
```
- /attachment ca
- Attachment ca, All units:
- Attachment [ca] used by 1 pieces:
- ..Piece [Iron Will counter] Attachment [ca] has 0 attachments:
```
The `ca` attachment is restored, along with the command, if the black/grey "Iron Will" token is moved and placed back in its current location.

The attachment trait:
<img width="949" alt="image" src="https://github.com/vassalengine/vassal/assets/58135975/98288102-4260-45f5-8671-ed4c36cc1ac8">

Game file (will open using Beta8 module shared in opening comment): 
[test IW attach.zip](https://github.com/vassalengine/vassal/files/15233460/test.IW.attach.zip)

**Comment by riverwanderer** [comment]
PR #13439 tested OK against both cases detailed above. Test for case 1 was to use the P001 Schleiz scenario, verifying that that problem does occur under 3.7.12 but the same log file shows fine under this fix. Thank you, @BrentEaston.

**Comment by BrentEaston** [comment]
Merged into 3.7.13
